### ISSUE TITLE
delete branch

### ISSUE THREAD

**Issue by moos3** [PagerDuty/lita-github#45]
It would be awesome if the bot could delete a given branch in a repo. I love the functionality but that's the only function I have to go to github manually to do.

**Comment by ericboehs** [comment]
This too is my only issue.

**Comment by theckman** [comment]
Hey there. If either of you two (@moos3 / @ericboehs)  would like to tackle implementing this I'll try to help out if I can. I no longer work at PagerDuty, nor do I use Lita, so I'd be a bit limited in what I could do. But I'd be happy to discuss implementation ideas / do any sort of code reviews.

If a PR gets opened, I'm certain someone internal to PagerDuty will ensure it gets properly reviewed and merged.

For implementation, it's in a bit of a weird place within the GitHub API. It's not considered an action on a Repository, but instead an action taken on `git data` which puts it in a different section of the API. The actions can do more than delete just branches, but anything that gets returned by this call:

* https://developer.github.com/v3/git/refs/#get-all-references

Looking at how GitHub is structuring their API it makes sense, but it throws a bit of a wrench in how this plugin was designed. Should the deletion command go within the `repo` subcommand with the details of what it's doing under the hood abstracted away, or should a new `git` subcommand be created?

I'm torn. I like going with the latter because it then looks like the GitHub API, but it then puts a burden on the user because they may need to know more about the GitHub API. I like keeping it in-line with the GitHub API because it means less questions of where to implement new functionality.

Maybe if we look at what the commands look like it'll sway our opinions:

```
!gh git ref del PagerDuty/lita-github heads/example_branch
!gh git ref delbranch PagerDuty/lita-github example_branch
!gh repo delbranch PagerDuty/lita-github example_branch
```

The one idea this gives me, assuming it's possible, would be to implement the `git ref` subcommand and then have `gh repo delbranch` wrap the proper invocation for `git ref
Maybe we implement the `git` subcommand, with the `ref` functionality fully handled, and we can then create `repo delbranch` which just aliases `gh git

**Comment by ericboehs** [comment]
Personally I don't care to delete branches independently. I would just like an option to auto delete on merge.

I could potentially PR this but it's not highly important to my team currently.

**Comment by ericboehs** [comment]
I did some research.

lita-github uses octokit to interface with GitHub. The pull request merge is done in the [github_pr handler](https://github.com/PagerDuty/lita-github/blob/master/lib/lita/handlers/github_pr.rb#L97).

In order to delete the branch automatically I'd add a call to [`octokit.delete_branch(branch)`](https://github.com/octokit/octokit.rb/blob/eedd2a30516824f564108cfb06b39f5d2fdb092f/lib/octokit/client/refs.rb#L99-L101) just after line 97 in github_pr.rb (conditionally on user config).

lita-github is also tested so there's probably a few tests to add in the 600+ line [github_pr_spec](https://github.com/PagerDuty/lita-github/blob/master/spec/unit/lita/handlers/github_pr_spec.rb).

**Comment by theckman** [comment]
Yeah. Honestly, would probably be 30-50 lines of code at-most. This includes the user config stuff, the extra calls / error handling, but most it would be tests.
